src/cdomains/pthreadDomain.ml

```diff
@@ -0,0 +1,96 @@
+open Prelude
+
+(** Thread ID *)
+module Tid = IntDomain.Flattened
+
+(** Context hash for function calls *)
+module Ctx = IntDomain.Flattened
+
+(** Set of predecessor nodes *)
+module Pred = struct
+  module Base = Basetype.ProgLocation
+  include SetDomain.Make (Base)
+
+  let of_node = singleton % MyCFG.getLoc
+
+  let of_current_node () = of_node @@ Option.get !MyCFG.current_node
+
+  let string_of_elt = Basetype.ProgLocation.show
+end
+
+type domain =
+  { tid : Tid.t
+  ; pred : Pred.t
+  ; ctx : Ctx.t
+  }
+[@@deriving to_yojson]
```

Comment:
Why defined outside of `D`? Right now `to_yojson` is being derived twice for this type.